apps/dav/lib/Connector/Sabre/File.php

```diff
@@ -242,7 +251,7 @@ public function put($data) {
 			if (isset($_SERVER['CONTENT_LENGTH']) && $_SERVER['REQUEST_METHOD'] === 'PUT') {
 				$expected = (int)$_SERVER['CONTENT_LENGTH'];
 				if ($count !== $expected) {
-					throw new BadRequest('Expected filesize of ' . $expected . ' bytes but read (from Nextcloud client) and wrote (to Nextcloud storage) ' . $count . ' bytes. Could either be a network problem on the sending side or a problem writing to the storage on the server side.');
+					throw new BadRequest($this->l10n->t('Expected filesize of %1$d bytes but read (from Nextcloud client) and wrote (to Nextcloud storage) %2$d bytes. Could either be a network problem on the sending side or a problem writing to the storage on the server side.', [$expected, $count]));
```

Comment:
I think this needs to use plurals than:
https://docs.nextcloud.com/server/latest/developer_manual/basics/front-end/l10n.html#correct-plurals

Reply:
To avoid a bunch of nested if else statements and duplication by keeping the sentence together (i.e. if first is plural and second is singular, first is singular and second is plural, both singular, et cetera) is it acceptable to split the sentence as below?
```php
$this->l10n->n(
  'Expected filesize of %n byte. ',
  'Expected filesize of %n bytes. ',
  $expected,
) . $this->l10n->n(
  'But read (from Nextcloud client) and wrote (to Nextcloud storage) %n byte. Could either be a network problem on the sending side or a problem writing to the storage on the server side.',
  'But read (from Nextcloud client) and wrote (to Nextcloud storage) %n bytes. Could either be a network problem on the sending side or a problem writing to the storage on the server side.',
  $bytesWritten,
)
```

Reply:
I guess the easiest would be:
```suggestion
					throw new BadRequest($this->l10n->t('Expected filesize of %1$s but read (from Nextcloud client) and wrote (to Nextcloud storage) %2$s. Could either be a network problem on the sending side or a problem writing to the storage on the server side.', [$this->l10n->n('%n byte', '%n bytes', $expected), $this->l10n->n('%n byte', '%n bytes', $count)]));
```

Splitting the sentence comes with it's own problem in RTL languages I think.

Reply:
Did that, I suppose splitting only the bytes part will be fine for translations